### PR TITLE
feat: support prop model deploy (Diff PR#4467 - PR#1430)

### PR DESCRIPTION
**Pull request by Captainia**
This PR is the diff of https://github.com/aws/sagemaker-python-sdk/pull/4467 and previous one https://github.com/aws/sagemaker-python-sdk-staging/pull/1430 for easy review. Once this is approved we will merged the official PR into public repo master branch.

### CODE REVIEW

File: src/sagemaker/jumpstart/factory/model.py [hunk 1]
```diff
@@ -199,7 +200,14 @@ def _add_instance_type_to_kwargs(
...
 
 def _add_image_uri_to_kwargs(kwargs: JumpStartModelInitKwargs) -> JumpStartModelInitKwargs:
-    """Sets image uri based on default or override, returns full kwargs."""
+    """Sets image uri based on default or override, returns full kwargs.
+
+    Uses placeholder image uri for JumpStart proprietary models that uses ModelPackages
+    """
+
+    if kwargs.model_type == JumpStartModelType.PROPRIETARY:
+        kwargs.image_uri = ""
```
Comment (evakravi) [inline src/sagemaker/jumpstart/factory/model.py]: can we set this to `None`?

File: tests/integ/sagemaker/jumpstart/estimator/test_jumpstart_estimator.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 from sagemaker.jumpstart.utils import get_jumpstart_content_bucket
 
 
-MAX_INIT_TIME_SECONDS = 5
+MAX_INIT_TIME_SECONDS = 15
```
Comment (evakravi) [inline tests/integ/sagemaker/jumpstart/estimator/test_jumpstart_estimator.py]: i'm curious, why was this needed?

File: tests/integ/sagemaker/jumpstart/estimator/test_jumpstart_estimator.py [hunk 3]
```diff
@@ -234,5 +234,3 @@ def test_instatiating_estimator(mock_warning_logger, setup):
     elapsed_time = time.perf_counter() - start_time
 
     assert elapsed_time <= MAX_INIT_TIME_SECONDS
-
-    mock_warning_logger.assert_called_once()
```
Comment (evakravi) [inline tests/integ/sagemaker/jumpstart/estimator/test_jumpstart_estimator.py]: why is this removed? i think the intention is we want a single warning log cause we're using a wildcard version

File: tests/integ/sagemaker/jumpstart/model/test_jumpstart_model.py [hunk 4]
```diff
@@ -262,3 +263,25 @@ def test_jumpstart_model_register(setup):
...
+
+@pytest.mark.skipif(
+    True,
+    reason="Only enable if test account is subscribed to the proprietary model",
+)
+def test_proprietary_jumpstart_model(setup):
+
+    model_id = "ai21-jurassic-2-light"
+
+    model = JumpStartModel(
```
Comment (evakravi) [inline tests/integ/sagemaker/jumpstart/model/test_jumpstart_model.py:277]: can you add the JumpStart tags here? Otherwise it won't be marked for deletion and we'll hit resoruce limits?

File: tests/unit/sagemaker/jumpstart/model/test_model.py [hunk 5]
```diff
@@ -441,6 +442,63 @@ def test_eula_gated_conditional_s3_prefix_metadata_model(
...
+        mock_session.return_value = sagemaker_session
+
+        model = JumpStartModel(
+            model_id=model_id,
+        )
+
+        mock_model_init.assert_called_once_with(
+            image_uri="",
+            model_data="s3://jumpstart-cache-prod-us-west-2/None",
+            source_dir="s3://jumpstart-cache-prod-us-west-2/None",
```
Comment (evakravi) [inline tests/unit/sagemaker/jumpstart/model/test_model.py]: this doesn't seem right. it should be:
```
image_uri=None,
model_data=None,
source_dir=None,
```

File: tests/unit/sagemaker/jumpstart/model/test_model.py [hunk 6]
```diff
@@ -441,6 +442,63 @@ def test_eula_gated_conditional_s3_prefix_metadata_model(
...
+
+        mock_validate_model_id_and_get_type.return_value = JumpStartModelType.PROPRIETARY
+        model_id, _ = "ai21-summarization", "*"
+
+        mock_get_model_specs.side_effect = get_spec_from_base_spec
+
+        mock_session.return_value = sagemaker_session
+
+        model = JumpStartModel(
+            model_id=model_id,
```
Comment (evakravi) [inline tests/unit/sagemaker/jumpstart/model/test_model.py]: what happens if you supply a version?

File: tests/unit/sagemaker/jumpstart/model/test_model.py [hunk 7]
```diff
@@ -441,6 +442,63 @@ def test_eula_gated_conditional_s3_prefix_metadata_model(
...
+
+        model.deploy()
+
+        mock_model_deploy.assert_called_once_with(
+            initial_instance_count=1,
+            instance_type="ml.p4de.24xlarge",
+            wait=True,
+            tags=[
+                {"Key": JumpStartTag.MODEL_ID, "Value": "ai21-summarization"},
+                {"Key": JumpStartTag.MODEL_VERSION, "Value": "*"},
```
Comment (evakravi) [inline tests/unit/sagemaker/jumpstart/model/test_model.py]: is this the version we want necessarily?

File: tests/integ/sagemaker/jumpstart/estimator/test_jumpstart_estimator.py (hunk 2 above)
Comment (Captainia) [reply tests/integ/sagemaker/jumpstart/estimator/test_jumpstart_estimator.py]: replied here: https://github.com/aws/sagemaker-python-sdk/pull/4467#discussion_r1509102642.  I think I'll go with approach 2 as it solves both this question and the assert_warning comment below

File: tests/integ/sagemaker/jumpstart/model/test_jumpstart_model.py (hunk 4 above)
Comment (Captainia) [reply tests/integ/sagemaker/jumpstart/model/test_jumpstart_model.py:277]: Thanks, added

File: tests/unit/sagemaker/jumpstart/model/test_model.py (hunk 5 above)
Comment (Captainia) [reply tests/unit/sagemaker/jumpstart/model/test_model.py]: Actually only for image_uri we can't pass None if re-using the sagemaker Model class, updated for model_data and source_dir

File: tests/unit/sagemaker/jumpstart/model/test_model.py (hunk 6 above)
Comment (Captainia) [reply tests/unit/sagemaker/jumpstart/model/test_model.py]: Version is supported and required now, since we shouldn't select wildcard version based on non-semantic version.

File: tests/unit/sagemaker/jumpstart/model/test_model.py (hunk 7 above)
Comment (Captainia) [reply tests/unit/sagemaker/jumpstart/model/test_model.py]: Thanks, fixed, we now require version for prop models and it is logged correctly.
